ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/RecoverySocketUnitTest.java

```diff
@@ -0,0 +1,165 @@
+    private static int periodicRecoveryPeriodInitialState, recoveryBackoffPeriodInitialState;
+
+    private InetAddress recoveryManagerHost = null;
+    private int recoveryManagerPort = 0;
+    private RecoveryManager recoveryManager;
+
+    @BeforeClass
+    public static void getInitialState() {
+        socketRecoveryListenerInitialState = recoveryPropertyManager.getRecoveryEnvironmentBean().isRecoveryListener();
+        periodicRecoveryPeriodInitialState =  recoveryPropertyManager.getRecoveryEnvironmentBean().getPeriodicRecoveryPeriod();
+        recoveryBackoffPeriodInitialState = recoveryPropertyManager.getRecoveryEnvironmentBean().getRecoveryBackoffPeriod();
+    }
+
+    @AfterClass
+    public static void returnBackInitialState() {
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryListener(socketRecoveryListenerInitialState);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setPeriodicRecoveryPeriod(periodicRecoveryPeriodInitialState);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryBackoffPeriod(recoveryBackoffPeriodInitialState);
+    }
+
+    @Before
+    public void enableRecoveryListener() throws InterruptedException{
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryListener(true);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setPeriodicRecoveryPeriod(1);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryBackoffPeriod(1);
+        recoveryManager = RecoveryManager.manager();
+        // wait for the socket would be established
+        Thread.sleep(50);
+    }
+
+    @After
+    public void returnStateOfRecoveryListener() {
+        try {
+            recoveryManager.terminate();
+        } catch (IllegalStateException ise) {
+            // cannot terminate cleanly which may happen because some of the tests which terminated recovery manager before
+            log.debugf(ise,"Cannot terminate recovery manager. This is probably not a problem as the test may stopped it already. Check the prior errors.");
+        }
+    }
+
+    @Test
+    public void socketPing() throws Exception {
+        try (Socket connectorSocket = getSocket()) {
+            // streams to and from the RecoveryManager listener
+            BufferedReader fromServer = new BufferedReader(new InputStreamReader(connectorSocket.getInputStream(), StandardCharsets.UTF_8));
+            PrintWriter toServer = new PrintWriter(new OutputStreamWriter(connectorSocket.getOutputStream(), StandardCharsets.UTF_8));
+
+            toServer.println(RecoveryDriver.PING);
+            toServer.flush();
+            String stringResponse = fromServer.readLine();
+            assertEquals("Expecting the correct response string for command " + RecoveryDriver.PING, RecoveryDriver.PONG, stringResponse);
+        } catch (final SocketTimeoutException stex) {
+            failOnSocketTimeout(stex, RecoveryDriver.PING);
```

Comment:
Is there a test where we expect the socket to time out and we verify that?

Reply:
No, but I can add it. Just note that the socket which times out is the one on the client side as the remote socket is the server socket which waits for a new connections.